packages/core/jkl/_spacing.scss

```diff
@@ -5,119 +5,132 @@
-        "large-screen": $spacing-3xs,
-    ),
-    "2xs": (
-        "small-screen": $spacing-2xs,
-        "large-screen": $spacing-2xs,
-    ),
-    "xs": (
-        "small-screen": $spacing-xs,
-        "large-screen": $spacing-xs,
-    ),
-    "s": (
-        "small-screen": $spacing-s,
-        "large-screen": $spacing-s,
-    ),
-    "m": (
-        "small-screen": $spacing-m,
-        "large-screen": $spacing-m,
-    ),
-    "l": (
-        "small-screen": $spacing-m,
-        "large-screen": $spacing-l,
-    ),
-    "xl": (
-        "small-screen": $spacing-l,
-        "large-screen": $spacing-xl,
-    ),
-    "2xl": (
-        "small-screen": $spacing-xl,
-        "large-screen": $spacing-2xl,
-    ),
-    "3xl": (
-        "small-screen": $spacing-2xl,
-        "large-screen": $spacing-3xl,
-    ),
-    "4xl": (
-        "small-screen": $spacing-3xl,
-        "large-screen": $spacing-4xl,
-    ),
+$spacing: (
+    "2": $spacing-2,
+    "4": $spacing-4,
+    "8": $spacing-8,
+    "12": $spacing-12,
+    "16": $spacing-16,
+    "24": $spacing-24,
+    "32": $spacing-32,
+    "40": $spacing-40,
+    "64": $spacing-64,
+    "104": $spacing-104,
+    "168": $spacing-168,
+    // Gamle t-skjortestørrelser for kompatibilitet
+    "3xs": $spacing-3xs,
+    "2xs": $spacing-2xs,
+    "xs": $spacing-xs,
+    "s": $spacing-s,
+    "m": $spacing-m,
+    "l": $spacing-l,
+    "xl": $spacing-xl,
+    "2xl": $spacing-2xl,
+    "3xl": $spacing-3xl,
+    "4xl": $spacing-4xl,
 );
 
-$positions: (top, bottom, left, right);
-
-@mixin spacing($spacing-name) {
-    // Separate spacing/position into spacing and position variables:
-    $start-index: string.index($spacing-name, "/") or 0;
-    $end-index: string.index($spacing-name, "/") or 9999999;
-    $spacing: string.slice($spacing-name, 0, $start-index - 1);
-    $position: string.slice($spacing-name, $end-index + 1);
+/// Her defineres par av spacinger for dynamiske spacing-klasser, som f.eks. .jkl-spacing-16-24
+/// @type Map
+$spacing-pairs: (("16", "24"), ("24", "40"), ("32", "40"), ("40", "64"), ("64", "104"), ("104", "168"));
 
-    @if map.has-key($spacing-new, $spacing) {
-        $variants: map.get($spacing-new, $spacing);
+/// Posisjoner som kan benyttes for spacing
+/// @type Map
+$positions: (top, bottom, left, right);
 
-        @if string.length($position) == 0 {
-            margin: map.get($variants, "small-screen");
+/// Gir deg marginer basert på stegene i spacing-skalaen.
+/// @param $steps Enten navnet på ett steg i skalaen, eller navnet på tosteg i skalaen adskilt av "/".
+/// Dersom du oppgir to steg brukes det første på små skjermer og det andre på store.
+/// @param {"top" | "right" | "bottom" | "left"} $position Lar deg spesifisere margin for kun én av sidene
+/// @example
+/// .class {
+///     @include jkl.spacing("40/64");
+/// }
+/// @example
+/// .class {
+///     @include jkl.spacing("16", "bottom");
+/// }
+@mixin spacing($steps, $position: null) {
```

Comment:
Oh, forresten! Renaming av parameternavn er [breaking i Sass](https://sass-lang.com/documentation/at-rules/mixin#keyword-arguments). Tenker vi kanskje kan beholde det gamle navnet?

Reply:
Det finnes ingen `spacing`-mixin i dag, så er ikke breaking for noe annet enn forslaget til @henrikhermansen. Tenker kanskje det er innafor? 😅

Reply:
![](https://media4.giphy.com/media/NCjISbEPFxm48/giphy.gif?cid=ecf05e4767r6g7ej2a7j1xweaz9idvzc0im7jdj5g3ktxbwg&rid=giphy.gif&ct=g)